### PR TITLE
NOC-258: add horizontal virtualization

### PR DESCRIPTION
**Pull request by ketewan**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Comment by ketewan** [comment]
File packages/nc-gui/components/project/spreadsheet/views/Cell.vue -- is pure separation of component, without any refactoring. Moreover, I will create task to return it back and keep it as the header for more consistency

File: packages/nc-gui/components/virtualizedTable/VirtualizedTable.vue [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        v-if="data"
+        class="xc-row-table nc-grid backgroundColorDefault"
+      >
+        <tbody>
+          <tr
+            v-for="({ row: rowObj, rowMeta }, row) in data"
+            :key="row"
+            class="nc-grid-row"
+            :class="{
+              'nc-new-row': rowMeta.new,
```
Comment (Faradey27) [inline packages/nc-gui/components/virtualizedTable/VirtualizedTable.vue:14]: does vue support ternary operator? like [rowMeta.new ? '1' : '2']: true

File: packages/nc-gui/components/virtualizedTable/VirtualizedTable.vue [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  props: {
+    data: [Array, Object]
+  },
+  data() {
+    return {
+      cell: Cell
+    }
+  },
+  mounted() {
+    console.log('virtualized table')
```
Comment (Faradey27) [inline packages/nc-gui/components/virtualizedTable/VirtualizedTable.vue:51]: do we need this console in prod?

File: packages/nc-gui/patches/vue-virtual-scroll-list+2.3.3.patch [hunk 3]
```diff
@@ -0,0 +1,202 @@
+diff --git a/node_modules/vue-virtual-scroll-list/dist/index.js b/node_modules/vue-virtual-scroll-list/dist/index.js
```
Comment (Faradey27) [inline packages/nc-gui/patches/vue-virtual-scroll-list+2.3.3.patch:1]: wdyt about leaving some comments inside of the patch that explains:
 - what we patched (why we added this ref, callback)
 - why we patched it (what problem we are solving)
 - what is conditions of patch being remove (maybe there are some issues in this relation)

**Review by Faradey27 — COMMENTED** [review]
Hey :)

Besides Cell component PR looks good and I did not review Cell according to your comment.

Several questions:
- why we added lock file in package that we probably did not touch? packages/nocodb/package-lock.json?
- How I can test it locally? by default my noco is empty

File: packages/nc-gui/components/virtualizedTable/VirtualizedTable.vue (hunk 1 above)
Comment (ketewan) [reply packages/nc-gui/components/virtualizedTable/VirtualizedTable.vue:14]: This file shouldn't exist at all, just forgot to remove it 😅

**Review by Faradey27 — APPROVED** [review]
Feels pretty laggy and ustable, but probably better than current state

**Pull request closed without merging**